ui/app/templates/jobs/run/index.hbs

```diff
@@ -6,5 +6,11 @@
 <Breadcrumb @crumb={{hash label="Run" args=(array "jobs.run")}} />
 {{page-title "Run a job"}}
 <section class="section">
+  {{#if (and this.sourceString (not this.disregardNameWarning))}}
+    <Hds::Alert @type="inline" @color="warning" data-test-job-name-warning as |A|>
+      <A.Title>Don't forget to change the job name!</A.Title>
+      <A.Description>Since you're cloning a job version's source as a new job, you'll want to change the job name. Otherwise, this will appear as a new version of the original job, rather than a new job.</A.Description>
+    </Hds::Alert>
+  {{/if}}
```

Comment:
It might be nice to swap out the job name with an empty string to prevent accidental overwrites. Unfortunately we don't a JS parser for HCL2. My first pass at a regexp was something like `^job\s+"(.*)"\s+\{` but some quick testing finds some corner cases, so maybe that's not such a good idea after all. If you've got an obvious way to do this, I'd go for it, otherwise we can skip.